src/core/server/elasticsearch/cluster_client.ts

```diff
@@ -197,4 +217,16 @@ export class ClusterClient {
       throw new Error('Cluster client cannot be used after it has been closed.');
     }
   }
+
+  private getHeaders(
+    request?: KibanaRequest | Request | FakeRequest
+  ): Record<string, string | string[] | undefined> {
+    if (!isRealRequest(request)) {
```

Comment:
when migrate from Legacy to New platform, we  don't need to support `Request` as a valid input and can switch to 
```javascript
if(isKibanaRequest(request)){
  const authHeaders = this.getAuthHeaders(request);
  const headers = toRawRequest(request).headers;
  return { ...headers, ...authHeaders };
} else {
  return request.headers;
}
```